stately/src/otherNativeMain/kotlin/co/touchlab/stately/concurrency/Lock.kt

```diff
@@ -1,9 +1,20 @@
 package co.touchlab.stately.concurrency
 
 import kotlin.native.concurrent.AtomicInt
+import kotlin.native.concurrent.ThreadLocal
 
+@ThreadLocal
+private object Thread {
+  val id: Int by lazy { Any().hashCode() }
```

Comment:
This relies on `Any().hashCode()` to be unique? Looking at the native impl, it'll use the object pointer val, which will be unique unless you get a recycled memory block. Unlikely but no impossible. Why not use a global AtomicInt and increment?

Reply:
Also, looking at the lock code, it relies on 0 as the indicator of "none", which may never be returned in practice as the id val here, but there's nothing about the hashCode contract that says it couldn't.

Reply:
Hi Kevin, I got the inspiration from the internal kotlin-native lock implementation which does essentially the same thing. But I agree, does seem to have some opportunity for failure.

I like the idea of a global counter. I think this would work:
```
private object GlobalThreadCounter {
  val threadCount = AtomicInt(0)
}

@ThreadLocal
private object Thread {
  val id: Int by lazy { GlobalThreadCounter.threadCount.addAndGet(1) }
}
```

Reply:
Interesting, this? https://github.com/JetBrains/kotlin-native/blob/master/runtime/src/main/kotlin/kotlin/native/concurrent/Lock.kt#L11

Maybe there's a reason that's ok that I'm not thinking of. Seems like a bad idea, though. Tempted to try to prove a bug in native

Reply:
Ugh. I'm slow today. That's fine because they're keeping the object ref. Taking a dash of that and releasing would be dangerous

Reply:
Ah. Okay. I'll fix that up to keep the reference of the current object then.

Reply:
Oooh yeah good point. Works if you keep the object around.